server/app/controllers/applicant/RedirectController.java

```diff
@@ -78,6 +84,13 @@ public CompletionStage<Result> programByName(Http.Request request, String progra
               }
 
               Applicant applicant = applicantFuture.join();
+              ApplicantData data = applicant.getApplicantData();
```

Comment:
Reading this method - it's not clear why we are setting preferred locale specifically here, in "programByName" action. To be setting locale sounds like something general, applied to various "entries" to the app.

Reply:
this is deep link case. applicant would get a link directly to that program and this is the method that would be called. agree that the naming is not super clear.